pkg/catalog/xds_certificates.go

```diff
@@ -40,10 +41,15 @@ func (mc *MeshCatalog) GetServiceFromEnvoyCertificate(cn certificate.CommonName)
 		return nil, err
 	}
 
-	return &service.MeshService{
-		Namespace: cnMeta.Namespace,
-		Name:      services[0].Name,
-	}, nil
+	for _, svc := range services {
+		meshService := service.MeshService{
+			Namespace: cnMeta.Namespace,
+			Name:      svc.Name,
+		}
```

Comment:
Just a thought, it might be a good idea to abstract the conversion of a `service` to `meshService` in a utils function and they use it everywhere. WDYT ?

Reply:
From kubernetes service to meshService? I think that's a good idea. Can you make an issue and tag it as "good first issue"?

Reply:
There is a helper for converting K8s service to MeshService in `catalog/routes.go`: `k8sSvcToMeshSvc`

Reply:
@ksubrmnn please could we use `k8sSvcToMeshSvc` wherever possible